modules/product/src/main/java/com/opengamma/strata/product/etd/EtdIdUtils.java

```diff
@@ -32,6 +36,20 @@
  */
 public final class EtdIdUtils {
 
+  /**
+   * Separator that is always present between the contract details and expiry + option details.
+   * <p>
+   * Example separator "-202304" in "F-IFEN-ABC-202304" or "O-IFEN-AB-CD-202304-PM12.34-U202309"
+   */
+  private static final String GROUPS_SEPARATOR = "-(?=\\d{6})";
+  private static final String CONTRACT_DETAILS_REGEX_GROUP_NAME = "contractDetails";
+  private static final String EXPIRY_AND_OPTION_DETAILS_GROUP_NAME = "expiryAndOptionDetails";
+  private static final Pattern SECURITY_ID_PATTERN = Pattern.compile(Messages.format(
+      "^(?<{}>.*){}(?<{}>.*)$",
+      CONTRACT_DETAILS_REGEX_GROUP_NAME,
+      GROUPS_SEPARATOR,
+      EXPIRY_AND_OPTION_DETAILS_GROUP_NAME));
```

Comment:
I tried making this as readable as possible so any other suggestions are more than welcome